File: 21560 - Sai Suraj/week 2/946_Validate Stack Sequences.py
class Solution:
    def validateStackSequences(self, pushed: List[int], popped: List[int]) -> bool:
        stack = []
        j =0 
        for i in range(len(pushed)):
            stack.append(pushed[i])
            while stack and j<len(popped) and stack[-1]==popped[j]:
                stack.pop()
                j+=1
            
        return j==len(popped) # or len(stack)==0

File: 21558 - Nithin/week 2/946-validate-stack-sequences/946-validate-stack-sequences.cpp
class Solution {
public:
    bool validateStackSequences(vector<int>& pushed, vector<int>& popped) {
        stack<int> stck;
        int i=0,j=0;
        int len = pushed.size();
        
        while(j < len){ 
             
            if(!stck.empty() && popped[j] == stck.top()){
                stck.pop();
                ++j;
            }
            
            
            else if(i < len && pushed[i] != popped[j]){
                stck.push(pushed[i]);
                ++i;
            }
            
            else if(i < len && pushed[i] == popped[j]){
                    ++i;
                    ++j;
            }
            
            else
                break;
        }
        
        return stck.empty();
    }
};
